pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerInterceptors.java

```diff
@@ -62,9 +63,9 @@ public ConsumerInterceptors(List<ConsumerInterceptor<T>> interceptors) {
      */
     public Message<T> beforeConsume(Consumer<T> consumer, Message<T> message) {
         Message<T> interceptorMessage = message;
-        for (int i = 0; i < interceptors.size(); i++) {
+        for (ConsumerInterceptor<T> interceptor : interceptors) {
```

Comment:
The reason for using int index was to avoid creating the iterator object (because then it needs to be GCed)

Reply:
Sorry, I tend to focus in readability in this kind of cases. Instead of previous for by index, what do you think of optimized index size for loop (prevents computing .size() for each iteration) ? :

```java
for (int i = 0, interceptorsSize = interceptors.size(); i < interceptorsSize; i++) {}
```

If you still don't agree I will rollback this :+1:

Reply:
Yes, generally we should favor readability but these hooks are being called in the hot path for each messages produced/consumed. Also, in most cases people will not have interceptors configured, so the overhead should be (close to) 0.

> what do you think of optimized index size for loop (prevents computing .size() for each iteration) ? :

Sure, that looks good

Reply:
Totally agree, done.